### PR TITLE
Refactor resize to accept string as an argument

### PR DESCRIPTION
**Pull request by patrick-tolosa**
#### What does this PR solve?
This PR allows all the resize functions to accept strings as well as numbers.
An attempt was made to reuse all the type definitions instead of adding changing them one by one.

#### Commit Changes:
* Create a type for all resize actions (resizeActionFactory type)
* Consolidate all type definitions to the factory type
* Remove all type defintions from the resize functions
* Change all docstrings to be number|string
* Add a Types namespace to put all our types
* Remove of the IResizeAction (It seemed redundant), if we'll need it in the future we can re-add it but for now it looks like it's pointless

We still have an open question on how to reuse all the "param" definitions

#### Final checklist
- [x] JSdoc - Add proper docstrings based on our PHP implementation
- [x] JSdoc - Hide private functions using @private
- [x] Implementation is aligned with PHP Reference(If different, please specify why)
- [x] Tests - Add proper tests to the added code

### CODE REVIEW

**Review by strausr — APPROVED** [review]

File: __TESTS__/unit/actions/Resize.test.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import CloudinaryConfig from "../../../src/config/CloudinaryConfig";
 import {AutoGravity} from "../../../src/constants/gravityObjects/GravityObjects";
 import Gravity from "../../../src/params/gravity/Gravity";
 import * as GravityObjects from '../../../src/constants/gravityObjects/GravityObjects';
-import {IResizeAction} from "../../../src/actions/resize/IResizeAction";
+import {IResizeAction} from "../../../src/actions/resize/IResizeTypes";
```
Comment (maoznir) [inline __TESTS__/unit/actions/Resize.test.ts:7]: the descriptions says this was removed?

File: src/actions/resize/IResizeTypes.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import ResizeAction from "./ResizeAction";
+
+/**
+ * @memberOf Types
+ * @name resizeActionFactory
+ * @description
+ * This is the type for all resize action methods (scale, fit, crop, etc.).
+ */
+export type resizeActionFactory = (width?:number|string, height?:number|string) => ResizeAction
```
Comment (maoznir) [inline src/actions/resize/IResizeTypes.ts:9]: Why the word "factory" do we use this for other builders?

Comment (patrick-tolosa) [reply src/actions/resize/IResizeTypes.ts:9]: This is a good question.
Strictly speaking, these functions are factories that return a Builder Class for an action.

Resize has a lot of these functions, and they're all reused, so it made sense to extract all these declarations into one place.

We could create these types for all of our action functions, but if they don't share the signature then it makes no sense.

The idea was the share the signature of the resizeAction(factory)
What do you think?

File: __TESTS__/unit/actions/Resize.test.ts (hunk 1 above)
Comment (patrick-tolosa) [reply __TESTS__/unit/actions/Resize.test.ts:7]: The interface for ResizeAction was removed (initially we had an interface for every action, but it wasn't really needed).

We do have some Types left though that are needed. (If we decide to keep the resizeActionFactory)

Comment (patrick-tolosa) [reply __TESTS__/unit/actions/Resize.test.ts]: Good catch, removed.

File: src/actions/resize/IResizeTypes.ts (hunk 2 above)
Comment (maoznir) [reply src/actions/resize/IResizeTypes.ts:9]: As long as it's internal only and does not affect the user im ok with it

**Review by maoznir — APPROVED** [review]

**Pull request merged**